quickwit/quickwit-janitor/src/retention_policy_execution.rs

```diff
@@ -70,145 +80,36 @@ pub async fn run_execute_retention_policy(
-        update_timestamp: i64,
-        publish_ts_opt: Option<i64>,
-        time_range: Option<RangeInclusive<i64>>,
-    ) -> Split {
-        Split {
-            split_metadata: SplitMetadata {
-                split_id: "foo".to_string(),
-                footer_offsets: 5..20,
-                time_range,
-                ..Default::default()
-            },
-            split_state: SplitState::Published,
-            update_timestamp,
-            publish_timestamp: publish_ts_opt,
-        }
+/// Get rid of splits without time range.
+///
+/// For retention policy with cutoff_reference set to `split_timestamp_field`, we cannot
+/// apply the retention policy to splits without time range.
+/// We instead list them in a warning message.
+fn ignore_not_applicable_splits(
+    index_id: &str,
+    splits: Vec<SplitMetadata>,
+    cutoff_reference: RetentionPolicyCutoffReference,
+) -> Vec<SplitMetadata> {
+    if cutoff_reference != RetentionPolicyCutoffReference::SplitTimestampField {
+        return splits;
     }
 
-    #[test]
-    fn test_is_split_expired_with_publish_timestamp() {
-        let one_hour_retention_policy = RetentionPolicy::new(
-            "1 hour".to_string(),
-            RetentionPolicyCutoffReference::PublishTimestamp,
-            "hourly".to_string(),
-        );
-
-        // A newly created split.
-        let now_timestamp = Utc::now().timestamp();
-        let split = make_split(now_timestamp, Some(now_timestamp), None);
-        assert!(!is_split_expired(
-            OffsetDateTime::now_utc(),
-            &split,
-            &one_hour_retention_policy
-        )
-        .unwrap());
-
-        let two_hours_ago_timestamp = (Utc::now() - Duration::hours(2)).timestamp();
-
-        // A two hours old split.
-        let split = make_split(two_hours_ago_timestamp, Some(two_hours_ago_timestamp), None);
-        assert!(is_split_expired(
-            OffsetDateTime::now_utc(),
-            &split,
-            &one_hour_retention_policy
-        )
-        .unwrap());
-
-        // A two hours old split without publish_timestamp
-        let split = make_split(two_hours_ago_timestamp, None, None);
-        assert!(is_split_expired(
-            OffsetDateTime::now_utc(),
-            &split,
-            &one_hour_retention_policy
-        )
-        .unwrap());
+    let mut expired_splits = Vec::with_capacity(splits.len());
+    let mut not_applicable_splits = vec![];
+    for split_metadata in splits {
+        if split_metadata.time_range.is_some() {
+            expired_splits.push(split_metadata);
+        } else {
+            not_applicable_splits.push(split_metadata);
+        }
     }
 
-    #[test]
-    fn test_is_split_expired_with_time_range() {
-        let one_hour_retention_policy = RetentionPolicy::new(
-            "1 hour".to_string(),
-            RetentionPolicyCutoffReference::SplitTimestampField,
-            "hourly".to_string(),
+    if !not_applicable_splits.is_empty() {
```

Comment:
Nitpick: I think a warning on each split is ok and that the usage of a filter will make the code shorter. Now we need two vectors, one for expired splits, and one for non applicable splits, as a reader, I would have prefered a shorter code on this.

Reply:
I thought it would too verbose to output every split as a new warning message and given that we are trying to avoid using so iterator pattern this made me choose to route.